cloud/defaults.go

```diff
@@ -230,6 +232,18 @@ func GetDefaultWindowsImage(k8sVersion string) (*infrav1.Image, error) {
 	return defaultImage, nil
 }
 
+// GetBootstrappingVMExtension returns the CAPZ Bootstrapping VM extension.
+// The CAPZ Bootstrapping extension is a simple clone of https://github.com/Azure/custom-script-extension-linux which allows running arbitrary scripts on the VM.
+// Its role is to detect and report Kubernetes bootstrap failure or success.
+func GetBootstrappingVMExtension(osType string, cloud string) (name, publisher, version string) {
+	// currently, the bootstrap extension is only available for Linux and in AzurePublicCloud.
+	if osType == "Linux" && cloud == azure.PublicCloud.Name {
+		return "CAPZ.Linux.Bootstrapping", "Microsoft.Azure.ContainerUpstream", "1.0"
+	}
+
```

Comment:
it's already published in all Azure public cloud regions. It's a clone of https://github.com/Azure/custom-script-extension-linux. I don't expect we'll have to maintain it further in the future since any changes to the script itself should happen in CAPZ, under the extension settings.